### ISSUE TITLE
scatter plot: false colors when using MarkerEdgeColor

### ISSUE THREAD

**Issue by twippenbeck** [matlab2tikz/matlab2tikz#899]
When using MarkerEdgeColor false colors are created after matlab2tikz-conversion with recent dev-version (see MWE + pictures below).

Whilst maybe related to #859 #861 #860 I do not see if these will solve the problem.

Is there a quick fix / workarround available? I have ~300 figures with custom colors to do, so modifying the tikz files by hand is quite ...

Thx!

MWE:

    figure;
    
    colors = [0 1 0; 0 0 1; 1 0 0];

    for iColor = 1:size(colors,1)
        X = rand(10,2);
        scatter(X(:,1),X(:,2),'o','MarkerEdgeColor',colors(iColor,:));
        hold all;
    end
    
    saveas(gcf,'MWE_scatter_EdgeColor_MatlabOrig.png')
    
    matlab2tikz(...
                'filename','MWE_scatter_EdgeColor.tex',...
                'encoding','UTF-8',...
                'standalone',true...
                );

MATLAB figure:
![mwe_scatter_edgecolor_matlaborig](https://cloud.githubusercontent.com/assets/14804753/15301120/6342de62-1bac-11e6-8301-3045c7dbc09f.png)

Output by M2T:
![mwe_scatter_edgecolor_matlab2tikz](https://cloud.githubusercontent.com/assets/14804753/15301128/6cd4b09a-1bac-11e6-9c06-367d79f1bbda.png)

**Comment by miscco** [comment]
Hm, as i understand it, there is a conflict between the cData and Markersize.

Have you tried to pass the color as the fourth input according to 
http://de.mathworks.com/help/matlab/ref/scatter.html

**Comment by miscco** [comment]
So regarding your problem i expect the culprit to be getScatterOptsOneColor

There we have the following code
```matlab
        if hasFaceColor && hasEdgeColor
            drawOptions = opts_add(drawOptions, 'draw', ecolor);
            drawOptions = opts_add(drawOptions, 'fill', xcolor);
        else
            drawOptions = opts_add(drawOptions, 'color', xcolor);
        end
```

So we default to the fill color if only one of the two is set. I dont really know if that is reasonable. Maybe we should set fill and draw seperately?

**Comment by twippenbeck** [comment]
1. fantastic! workaround  (color as the fourth input ) works perfectly for predefined and user defined colors (tested). you made my day :-)
2. fyi: I also used scenarios with `{'MarkerEdgeColor',xyz}` and `{'MarkerFaceColor','none'}` at the same time. Those did not work either

**Comment by miscco** [comment]
@twippenbeck Great that it worksfor you that way. Can you for the sake of clarity check #900? That should also fix the second case.

**Comment by miscco** [comment]
@twippenbeck As #900 is merged now can you please check if your problem is still there with the latest develop branch? As far as i can tell the secon scenario should be fixed by now.

**Comment by egeerardyn** [comment]
I can confirm this is fixed by #900 for both R2014a and R2014b.

Output looks like:
![mwe_scatter_edgecolor_matlaborig](https://cloud.githubusercontent.com/assets/177360/15317854/8aeec77e-1c23-11e6-9fe3-7679207378ff.png)
![mwe_scatter_edgecolor_tikz](https://cloud.githubusercontent.com/assets/177360/15317894/c0e03322-1c23-11e6-8b57-b493ec92489c.png)

**Comment by twippenbeck** [comment]
I'm sorry for the delayed response. I'm new to git and when I finally found out how to get miscco's commit locally I found all has been done ....
I confirm everything works fine now.
Thank you! @miscco @egeerardyn
